lib/workers/global/config/parse/file.ts

```diff
@@ -1,23 +1,60 @@
+import fs, { readFile } from 'fs-extra';
+import { load } from 'js-yaml';
+import JSON5 from 'json5';
 import upath from 'upath';
 import { migrateConfig } from '../../../../config/migration';
-import type { AllConfig } from '../../../../config/types';
+import type { AllConfig, RenovateConfig } from '../../../../config/types';
 import { logger } from '../../../../logger';
 
+export const fileDeprecationMessage =
+  'Providing a config file without an extension is deprecated. Please use explicit file extensions.';
```

Comment:
Could we instead just make this a documented change in the new major release?

Reply:
Ok that was also my thinking originally in https://github.com/renovatebot/renovate/pull/12644#discussion_r751254810, @viceice @rarkins, do you both agree here, and I go ahead with this and have no handling of files without extensions?

Reply:
@viceice can we issue a breaking change that the config file must have an extension?

Reply:
Sure. 👍

Reply:
I make it fail now and I've added some tests and a note in the docs.

I added a breaking change trailer but I noticed a typo :facepalm: but I guess you'll be squashing anyway :)